src/main/java/crawlercommons/sitemaps/sax/RSSHandler.java

```diff
@@ -109,47 +107,55 @@ public void startElement(String uri, String localName, String qName, Attributes
      */
     @Override
     public void endElement(String uri, String localName, String qName) throws SAXException {
-        if ("link".equals(currentElement()) || "guid".equals(currentElement())) {
+        if ("link".equals(localName)) {
             setLocURL();
-        } else if ("item".equals(currentElement())) {
+        } else if ("guid".equals(localName)) {
+            // accept as link if
+            // - a valid absolute URL (not a URN, UUID or similar)
+            // - and no <link> found yet
+            if (locURL == null) {
+                setLocURL();
+            }
+            resetCharacterBuffer();
+        } else if ("item".equals(localName)) {
             maybeAddSiteMapUrl();
-        } else if ("rss".equals(currentElement())) {
+        } else if ("rss".equals(localName)) {
             sitemap.setProcessed(true);
+        } else if ("pubDate".equals(localName)) {
+            String value = getAndResetCharacterBuffer();
+            if (value != null) {
```

Comment:
When would `value` ever be null? And is support for `"pubDate"` something new?

Reply:
> When would `value` ever be null?

Correct, will change it.

> And is support for `"pubDate"` something new?

The code has been moved from `characters()` to `endElement()`.